crates/sail-plan/src/resolver/data_source.rs

```diff
@@ -31,8 +32,28 @@ impl PlanResolver<'_> {
                     url
                 } else {
                     let store = self.ctx.runtime_env().object_store(&url)?;
+
                     if store.head(url.prefix()).await.is_ok() {
-                        url
+                        // HDFS returns okay if directory exists without trailing delimiter, but returning that causes an error when
+                        // listing files in the directory, hence the double check
```

Comment:
Could you link to the PR in the `hdfs-native-object-store` repository? Let's mention that the logic will no longer be needed once the fix is released in the dependency.

Reply:
Looks like they pushed a new version to crates.io yesterday which included the change. I removed the check and tested to confirm. It works as expected now.